### ISSUE TITLE
S4143 False positives with `List<T>.Add` method calls

### ISSUE THREAD

**Issue by cg110** [SonarSource/sonar-dotnet#2674]
### Description

Using the SonarAnalyzer.CSharp nuget, S4143 flags issues for List<String> (and probably any lists)

### Repro steps

Using the code:
```
List<string> S4143 = new List<string>();
S4143.Add("SomeText");
S4143.Add("\n");
S4143.Add("SomeText");
S4143.Add("\n");
```

The last 2 Adds are showing with warnings for S4143, as they are Adds they're extending the list rather than overwriting any existing entries.

### Expected behavior

The above code shouldn't generate a warning.

### Actual behavior

S4143 generates warnings.

### Known workarounds

Set the warning to info or suppress.

### Related information

* SonarC# Version: 7.16.0.8981
* Visual Studio Version: 2017, 15.9.16
* If running through the Scanner for MSBuild, its version

**Comment by Corniel** [comment]
I think it is okay that this warning is shown. You could argue that this behaviour should not be applied on a `List<string>`, but I think that adding string literals to a list this way is a bad practice.

You could argue that the message should than be different, but it tells you (indirectly) why this is a bad practice: a list is intended for keeping track of multiple elements of T (`string` in this case), not for building dynamic text. (Then you should use `StringBuilder`.)

**Comment by cg110** [comment]
I'd argue that a list is allowed to have duplicate entries in it, if I wanted unique then I'd be using a dictionary or hashset.

**Comment by christophe-zurn-sonarsource** [comment]
Hello @cg110 , thanks for reporting this.
Considering the rule, it is indeed a false positive, as the elements added later are not overriding the previous ones.
